services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/AwsConnectionServiceImpl.java

```diff
@@ -45,43 +44,40 @@ public AwsDatasourceResponse saveConnectionInfo(AwsDatasourceRequest awsDatasour
     }
 
     @Override
-    public AwsDatasourceResponse updateConnectionInfo(AwsDatasourceRequest awsDatasourceRequest, Integer id, Projects project) throws ConnectionNotFoundException {
+    public AwsDatasourceResponse updateConnectionInfo(AwsDatasourceRequest awsDatasourceRequest, Integer id, String userName) throws ConnectionNotFoundException {
+
+        AWSDatasource dataSource = awsConnectionRepo.findByIdAndActive(id,true).orElseThrow(() -> new ConnectionNotFoundException("Connection not found"));
+        Optional<AWSDatasource> dataSources = awsConnectionRepo.findByDataSourceNameIgnoreCaseAndActive(awsDatasourceRequest.getDataSourceName().trim(),true);
+        if (dataSources.isPresent()) {
+            throw new BadRequestException("Data source name already exist.");
+        }
 
-        AWSDatasource dataSource = awsConnectionRepo.findById(id).orElseThrow(() -> new ConnectionNotFoundException("Connection not found"));
         dataSource.setDataSourceName(awsDatasourceRequest.getDataSourceName());
-        dataSource.setModifiedBy(project.getModifiedBy());
-        dataSource.setModifiedDate(DateTime.now().toDate());
-        return createResponse(awsConnectionRepo.save(dataSource));
+        dataSource.setModifiedBy(userName);
+        return createResponse(awsConnectionRepo.saveAndFlush(dataSource));
     }
 
     private AwsDatasourceResponse createResponse(AWSDatasource awsDatasource) {
-        AwsDatasourceResponse response = new AwsDatasourceResponse();
-        response.setId(awsDatasource.getId());
-        response.setDataSourceName(awsDatasource.getDataSourceName());
-        response.setConnectionType(awsDatasource.getConnectionType());
+        ModelMapper mapper = new ModelMapper();
+        AwsDatasourceResponse response = mapper.map(awsDatasource,AwsDatasourceResponse.class);
         response.setIamRole(awsDatasource.getCredentialOrRole());
-        response.setCreatedDate(awsDatasource.getCreatedDate());
-        response.setModifiedDate(awsDatasource.getModifiedDate());
         return response;
     }
 
-
     @Override
-    public List<AwsDatasourceResponse> getAllConnection() {
+    public List<AwsDatasourceResponse> getActiveConnections(boolean active) {
         List<AwsDatasourceResponse> response = new ArrayList<>();
-        List<AWSDatasource> awsDatasource = awsConnectionRepo.findAll();
-
+        List<AWSDatasource> awsDatasource = awsConnectionRepo.findByActiveOrderByCreatedDateDesc(active);
         awsDatasource.forEach(datasource -> {
```

Comment:
Check if we need null check here

Reply:
Used Optional class to remove null pointer exception